resources/NBTest/NBTest_062_TestRouteHandler.py

```diff
@@ -424,3 +426,439 @@
+wrap_unwrap_2_processor = wrap_unwrap_processor_2.add_wrap_or_unwrap_trades_to_route(split_trade_instructions_2_split, route_struct_2, flashloan_struct_0)
+wrap_unwrap_3_processor = wrap_unwrap_processor_3.add_wrap_or_unwrap_trades_to_route(split_trade_instructions_3_split, route_struct_3, flashloan_struct_1)
+
+
+flashloan_struct_0=[{'platformId': 7, 'sourceTokens': ['0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'], 'sourceAmounts': [15000000000000000000]}]
+flashloan_struct_1=[{'platformId': 2, 'sourceTokens': ['0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'], 'sourceAmounts': [15000000000000000000]}]
+
+balances_0 = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 15000000000000000000, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 0, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_1 = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 0, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 15000000000000000000, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_2 = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 15000000000000000000, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 0, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_3 = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 0, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 15000000000000000000, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+
+balances_0_processor = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 15000000000000000000, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 0, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_1_processor = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 0, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 15000000000000000000, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_2_processor = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 15000000000000000000, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 0, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+balances_3_processor = {'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2': 0, '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE': 15000000000000000000, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': 0}
+
+
+def check_no_negative(balance_dict):
+    for tkn in balance_dict.keys():
+        assert balance_dict[tkn] >= 0, f"negative balance found: {balance_dict}"
+
+def loop_balances(_route_struct, _balance):
+    for trade in _route_struct:
+        #print(f"balance before trade {_balance}")
+        _balance[trade['sourceToken']] -= trade['sourceAmount']
+        _balance[trade['targetToken']] += trade['minTargetAmount']
+        check_no_negative(_balance)
+
+
+for idx, (_route, _balance) in enumerate([(wrap_unwrap_0_processor, balances_0_processor), (wrap_unwrap_1_processor, balances_1_processor), (wrap_unwrap_2_processor, balances_2_processor), (wrap_unwrap_3_processor, balances_3_processor)]):
+    #print(f"starting loop {idx}")
+    loop_balances(_route, _balance)
+
+## For Debugging
```

Comment:
All commented out code sections should be removed before we merge changes.

Reply:
Removed commented code